### DISCUSSION TITLE
Sugestões de alteração dos nomes das Primitivas

### DISCUSSION THREAD

**Discussion by VitBrandao** [DesignLiquido/delegua#142] [Ideias]
Conforme as implementações sugeridas na [discussão levantada pelo Alessandro](https://github.com/DesignLiquido/delegua/discussions/125) vão ganhando forma, gostaria de trazer aqui também algumas sugestões de **alteração dos nomes** de alguns dos métodos de texto e de vetor.   

Acredito que esta seja uma discussão mais conceitual do que técnica. Porém, pensando na experiência dos usuários e na proposta de **inclusão social da Delegua**, creio que não seja exagero refletirmos sobre como os métodos primitivos de Vetores e de Texto possam ter um valor semântico que informe ao usuário a finalidade de cada operação.

Meu objetivo aqui é apenas iniciar a conversa trazendo algumas sugestões. Agradeço a todos(as) que se sentirem a vontade tanto para concordar quanto para refutar.

### **1. fatia()**

Ao contrário dos demais métodos de vetor, esse método **não está no infinitivo**, o que pode passar a ideia de que seja um método de verificação (tal qual o **inclui()**). 

Acredito que **fatiar()** passe melhor a ideia de que uma ação está sendo executada, tal qual o **ordenar()**, o **juntar(),** etc.

Outras possíveis sugestões:
- **extrair()**;
- **fracionar()**;

------
### **2. empilhar()**

Para esse método, sugiro a alteração do nome para **adicionar()**, pois acredito que a palavra possua maior valor semântico para a operação de adicionar um novo item a uma lista. Além disso, o verbo 'adicionar' é sempre o termo utilizado para descrever o método **push()** nas **documentações**:

_Documentação do **TypeScript** (MDN):_
> "O push() adiciona um ou mais elementos ao final de um array e retorna o novo comprimento do array".

_Documentação do **JavaScript** (W3):_ 
> "O push() adiciona novos itens ao final de uma lista".

Em **Python**, o método em si se chamada **add().**:
> O add() adiciona um elemento ao conjunto.

------
### **3. subtexto**

Esse método de Texto utiliza o **slice()** do TypeScript tal qual o nosso método **fatia()** das primitivas de Vetor. 

A partir disso, sugiro **que utilizemos o mesmo nome** aqui (**fatia()** ou **fatiar()**).

Em Python, um dos recursos que acho muito interessante é o fato de podermos percorrer strings e listas **com a mesma facilidade**, pois os métodos avaliam os **caracteres** de uma string da mesma forma que avalia os **índices** de uma lista. Logo, acredito que utilizar o mesmo nome nas primitivas de Texto e de Vetor possa ser interessante, assim como já estamos fazendo com o método **inclui().**  

Considero que novas opiniões e sugestões possam ser construtivas para essa conversa. Agradeço desde já pela atenção.

**Comment by leonelsanchesdasilva** [comment]
# fatia()

Concordo com uniformizar para o infinitivo, `fatiar()`.

**Comment by leonelsanchesdasilva** [comment]
# empilhar()

JavaScript não possui uma estrutura de dados específica para pilha. Em estruturas de dados, quando estumados pilha, aprendemos que uma lista pode ser usada para implementar uma pilha. Creio que o nome `push()`, portanto, tenha vindo deste conceito. 

Defendo adicionar o método `adicionar()` como um sinônimo de `empilhar()`, mantendo as duas formas, já que Delégua, por ora, não possui uma estrutura específica de pilha.

**Comment by leonelsanchesdasilva** [comment]
# subtexto()

Concordo com a adição de `fatiar()` como sinônimo de `subtexto()`, mas discordo quanto à eliminação do método `subtexto()` da primitiva de texto. A razão são várias outras linguagens de programação que implementam um método chamado `substring()`. Como vejo como importante mantermos casos de equivalência entre Delégua e linguagens de programação em inglês, defendo manter `subtexto()` da forma em que está.
